numpy/_core/meson.build

```diff
@@ -646,6 +646,23 @@ src_ufunc_api = custom_target('__ufunc_api',
   install_tag: 'devel'
 )
 
+# Write out pkg-config file
+# -------------------------
+
+# Note: we can't use Meson's built-in pkgconfig module, because we have to
+# install numpy.pc within site-packages rather than in its normal location.
```

Comment:
This is technically not true because you can tell pkgconfig.generate() to use a specific install_dir.

Note that in order to handle relocatable wheels you'd then still want to define pkgconfig.relocatable=true inside cibuildwheel and then calculating it as offset from the wheel root instead of /usr would be a bit interesting to negotiate, because no one has actually tried to account for pkg-config files plus headers installed to something that isn't actually the --prefix. Hence it assumes that it needs to relativize everything based on prefix in order to recurse back down to includedir.

So you may prefer configure_file but not for the reasons this comment suggests. Or you may wish to propose changes to the pkgconfig module too. :)

Reply:
> Hence it assumes that it needs to relativize everything based on prefix in order to recurse back down to includedir.

Indeed. I did try to use `pkgconfig.generate()`, the problem is that there is no way to tell it where the include dir is.

> Or you may wish to propose changes to the pkgconfig module too.

 There's two things indeed:
- the docs assume that a library is the first input, it's unclear that header-only support is actually working
- `includedir` is now actually a reserved variable; to make this work that restriction should be dropped or a new keyword to set the include dir relative to the install dir should be added.